mm/ktsan/ktsan.c

```diff
@@ -299,7 +299,15 @@ void ktsan_thr_stop(void)
 void ktsan_thr_event_disable(void)
 {
 	ENTER(false, true);
-	kt_thr_event_disable(thr, pc);
+
+	if (kt_thr_event_disable(thr, pc)) {
+		/* Disable interrupts as well. Otherwise all events in
+		   interrupts that happen during that time will be ignored. */
+		thr->irq_flags_before_disable = kt_flags;
```

Comment:
Pass &kt_flags to kt_thr_event_disable, then move this code into kt_thr_event_disable.
This code is duplicated across several functions.